apis/controlplane/v1alpha4/nestedapiserver_types.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha4
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	addonv1alpha1 "sigs.k8s.io/kubebuilder-declarative-pattern/pkg/patterns/addon/pkg/apis/v1alpha1"
+)
+
+type NestedAPIServerPhase string
+
+const (
+	NestedAPIServerReady   NestedAPIServerPhase = "ready"
+	NestedAPIServerUnready NestedAPIServerPhase = "unready"
+)
+
+// NestedAPIServerSpec defines the desired state of NestedAPIServer
+type NestedAPIServerSpec struct {
+	// NestedComponentSpec contains the common and user-specified information that are
+	// required for creating the component
+	// +optional
+	NestedComponentSpec `json:",inline"`
+}
+
+// NestedAPIServerStatus defines the observed state of NestedAPIServer
+type NestedAPIServerStatus struct {
+	// Ready is set if all resources have been created
+	// +kubebuilder:default=false
+	Ready bool `json:"ready,omitempty"`
```

Comment:
Do we still want this given we have `phase`?

Reply:
Will remove this field from status